src/NServiceBus.Core/Routing/FileBasedDynamicRouting/AsyncTimer.cs

```diff
@@ -34,6 +34,10 @@ public void Start(Func<Task> callback, TimeSpan interval, Action<Exception> erro
 
         public Task Stop()
         {
+            if (tokenSource == null)
```

Comment:
did you ran into a scenario where the tokenSource was null? I think we don't stop FST which didn't run. Also, wouldn't it make more sense to check for `task`?